### PR TITLE
Integration test activemq

### PR DESCRIPTION
**Pull request by robsunday**
Primary goal of this PR is to use JMX Metric Insight to gather ActiveMQ JMX properties while preserving compatibility with JMX Gatherer.

At the moment there are two lines of ActiveMQ product:
- Classic
- Artemis

This test covers Classic line. Artemis line have different set of JMX attributes, so it should have separate yaml and test (nice summary of differences is here: https://www.datadoghq.com/blog/activemq-architecture-and-metrics/). Artemis support may be included in the future.

Assertions in the ActiveMQ integration test are copied from opentelemetry-java/contrib/jmx-metrics while activemq.yaml is taken from opentelemetry-java-instrumentation/jmx-metrics (JMX Metric Insight)
Minor changes were appiled to both of them. These changes are described as comments in the files.

### CODE REVIEW

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java:56]: [for reviewer] isMonotonic=false was added due to updatedDefinition of asserSumWithAttributes.

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java:67]: [for reviewer] isMonotonic=false was added due to updatedDefinition of asserSumWithAttributes.

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java:78]: [for reviewer] isMonotonic=false was added due to updatedDefinition of asserSumWithAttributes.

File: jmx-scraper/src/main/resources/activemq.yaml
Comment (robsunday) [inline jmx-scraper/src/main/resources/activemq.yaml:1]: Metric names has been changed to match that used in jmx-metrics

File: jmx-scraper/src/main/resources/activemq.yaml
Comment (robsunday) [inline jmx-scraper/src/main/resources/activemq.yaml:54]: This attribute was present in original activemq.yaml. Because of this some minor changes has to be made in assertions to check this metric attribute

File: jmx-scraper/src/main/resources/activemq/env
Comment (robsunday) [inline jmx-scraper/src/main/resources/activemq/env:1]: Once we merge it will be good to update resources folder structure. I did not want to do too much in this PR.

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java:26]: Updated activemq docker image to the most recent ActiveMQ Classic version (from rmohr/activemq:5.15.9 used in JMX Metric Gatherer)

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/MetricAssertions.java
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/MetricAssertions.java:80]: [for reviewer] This override method is added to support 'monotonic' flag and metric attributes at he same time.

File: jmx-scraper/src/main/resources/activemq.yaml
Comment (SylvainJuge) [reply jmx-scraper/src/main/resources/activemq.yaml:54]: Maybe add an explicit comment like in the `tomcat.yaml` to explain the difference, adding one extra metric attribute should not have much impact and thus it should remain compatible.

```suggestion
      # minor divergence from activemq.groovy to capture broker name, making it closer to
      # the definition in JMX Insights
      broker: param(brokerName)
```

File: jmx-scraper/src/main/resources/activemq.yaml
Comment (SylvainJuge) [reply jmx-scraper/src/main/resources/activemq.yaml:1]: You should probably update the PR description to reflect that the current goal is to preserve compatibility with `jmx-metrics` (JMX Gatherer).

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (SylvainJuge) [reply jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java:56]: For all of the 3 metrics that are non-monotonic here I think it was probably a bug in the original test of the groovy version. The fact that they all have `current` in their description means that they should represent a value that can go up or down, and thus can't be monotonic.

File: jmx-scraper/src/main/resources/activemq.yaml
Comment (SylvainJuge) [reply jmx-scraper/src/main/resources/activemq.yaml:1]: And also, only the classic activemq is supported, not the Artemis one.

**Review by SylvainJuge — APPROVED** [review]

**Comment by trask** [comment]
thanks, @robsunday mentioned on slack that he may push one more update, will wait for his confirmation

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (trask) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java:29]: can you add a comment why we need to customize activemq/env? thanks

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (robsunday) [reply jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java]: This is actually a very good point. Replacing file with JAVA_TOOL_OPTIONS was trivial and worked. '''env''' file was used in jmx-metrics and ported here, but really was not needed. 
Great suggestion!

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/TomcatIntegrationTest.java
Comment (robsunday) [inline jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/TomcatIntegrationTest.java:33]: This is not needed for Tomcat as well.

File: jmx-scraper/src/main/resources/activemq.yaml
Comment (SylvainJuge) [reply jmx-scraper/src/main/resources/activemq.yaml:13]: Yes, those are likely to remain "as is", and regarding their location having them in a dedicated folder might also make sense when we reorganize  but it's another topic.

I'm just trying to not to forget about this, but a TODO here is probably not the best choice, it's probably better to add this to the issue where we listed all the supported systems for feature parity: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/12158 so I've updated it to ensure we don't forget about it.

**Review by SylvainJuge — APPROVED** [review]

File: jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java
Comment (robsunday) [reply jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/ActiveMqIntegrationTest.java]: Finally I got rid of this file at all

**Review by trask — APPROVED** [review]

**Pull request merged**
